tests/cephadm/test_cephadm_ansible_wrapper.py

```diff
@@ -117,20 +116,21 @@ def run(ceph_cluster, **kwargs):
         if config.get("build_type") not in ["ga", "ga-async"]:
             extra_vars["image"] = config.get("container_image")
 
-    elif module in ["ceph_orch_host", "ceph_orch_apply"]:
-        osd_node = module_args.get("osd_node")
-        osd_node = get_node_by_id(nodes, osd_node)
+    elif module == "ceph_orch_apply":
         extra_vars["label"] = module_args.get("label")
 
-        if module == "ceph_orch_host":
-            extra_vars["osd_node"] = osd_node.hostname
-            extra_vars["ip_address"] = get_node_ip(nodes, osd_node.hostname)
-            CopyCephSshKeyToHost.run(installer, osd_node)
-
     elif module == "ceph_orch_host":
-        osd_node = module_args.get("osd_node")
-        extra_vars["osd_node"] = get_node_by_id(nodes, osd_node).hostname
-        extra_vars["ip_address"] = get_node_ip(nodes, osd_node)
+        node = module_args.get("host")
+        state = module_args.get("state")
+        label = module_args.get("label")
+        extra_vars["ip_address"] = get_node_ip(nodes, node)
+        node = get_node_by_id(nodes, node)
+        extra_vars["node"] = node.hostname
+        if state:
+            extra_vars["state"] = state
+        if label:
+            extra_vars["label"] = label
+        CopyCephSshKeyToHost.run(installer, node)
```

Comment:
This fuction need to execute only when new host is added. This should go else part of state if condition or need to check behaviour for `state` as `present`

Reply:
Noted. Changes made

Reply:
Adding note to handle this use case.